src/rdata.cpp

```diff
@@ -77,6 +75,9 @@ ReturnData::ReturnData(Solver const& solver, const Model *model)
     if (sensi >= AMICI_SENSI_ORDER_FIRST){
         sllh.resize(nplist, getNaN());
         sx0.resize(nx * nplist, getNaN());
+        FIM.resize(nplist * nplist, 0.0);
+        res.resize(nt * model->nytrue, 0.0);
+        sres.resize(nt * model->nytrue * nplist, 0.0);
```

Comment:
Depends on `sy`, shouldn't that be moved inside `if (sensi_meth == AMICI_SENSI_FSA ...` below?